.github/workflows/CI.yml

```diff
@@ -0,0 +1,82 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```

Comment:
I think it makes sense to run CI on all branches

would simplify to `on: [push, pull_request]` which I think is also a little more readable for PRs than the empty `pull_request:`

Reply:
This results in double runs, which results in wasting a lot of energy to run the whole build and test 12 times (twice the amount that is actually needed).